wp-content/themes/core/components/blocks/gallery_grid/Gallery_Grid_Controller.php

```diff
@@ -0,0 +1,346 @@
+				'c-block__title',
+				'b-gallery-grid__title',
+				'h3',
+			],
+			Text_Controller::CONTENT => $this->title ?? '',
+		] );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_slideshow_title() {
+		return $this->title ?? '';
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_content(): Deferred_Component {
+		return defer_template_part( 'components/container/container', null, [
+			Container_Controller::CLASSES => [
+				'c-block__description',
+				'b-gallery-grid__description',
+				't-sink',
+				's-sink',
+			],
+			Container_Controller::CONTENT => $this->description ?? '',
+		] );
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_slideshow_button() {
+		return [
+			Button_Controller::CLASSES    => [
+				'c-block__cta-link',
+				'a-btn',
+				'a-btn--has-icon-after',
+				'icon-arrow-right'
+			],
+			Button_Controller::ATTRS      => [ 'data-js'  => 'dialog-trigger', 'data-content' => 'dialog-content-' . $this->id ],
+			Button_Controller::CONTENT    => esc_html__( 'View slideshow', 'tribe' ),
+		];
+	}
+
+	/**
+	 * Get the Slider
+	 *
+	 * @return array
+	 */
+	public function get_slider_args(): array {
+		$main_attrs = [];
+
+		$swiper_options = '{"preloadImages":"false","lazy":"true","watchSlidesVisibility":"true","spaceBetween":60,"ally":"true","keyboard":"true","grabCursor":"true","navigation":{"nextEl":".swiper-button-next","prevEl":".swiper-button-prev"}}';
+
+		$main_attrs['data-swiper-options'] = $swiper_options;
+
+		$slider = [
+			Slider_Controller::SLIDES          => $this->get_slides(),
+			Slider_Controller::SHOW_CAROUSEL   => false,
+			Slider_Controller::SHOW_ARROWS     => true,
+			Slider_Controller::SHOW_PAGINATION => false,
+			Slider_Controller::MAIN_ATTRS      => $main_attrs,
+			Slider_Controller::CLASSES         => [ 'b-gallery-grid__slider' ],
+		];
+
+		return $slider;
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_gallery_img_ids(): array {
+		$gallery_ids = [];
+
+		if ( empty( $this->gallery ) ) {
+			return $gallery_ids;
+		}
+
+		foreach ( $this->gallery as $img ) {
+			if ( empty( $img ) ) {
+				return [];
+			}
+			$gallery_ids[] = $img[ 'id' ];
+		}
+
+		return $gallery_ids;
+	}
+
+	/**
+	 * @param int $slide_id
+	 *
+	 * @return string|Deferred_Component
+	 */
+	protected function get_image_caption( int $slide_id ) {
+		$thumbnail_image = get_posts( [ 'p' => $slide_id, 'post_type' => 'attachment' ] );
+
+		if ( empty( $thumbnail_image ) && empty( $thumbnail_image[0] ) ) {
+			return '';
+		}
+
+		return defer_template_part( 'components/text/text', null, [
+			Text_Controller::CLASSES => [
+				'b-gallery__slide-caption',
+			],
+			Text_Controller::CONTENT => esc_html( $thumbnail_image[0]->post_excerpt ) ?? '',
+		] );
+	}
+
+	/**
+	 * @param $img_id
+	 *
+	 * @return string
+	 */
+	protected function get_image_template( $index, $img_id ): string {
+		$img          = wp_get_attachment_image( $img_id, Image_Sizes::CORE_FULL );
```

Comment:
Hmm.... I'd prefer to see you use an actual image component here, along with a few image sizes (even just WP's `medium` & `large` in addition to the `CORE_FULL` size. That will better support mobile devices along with wrapping the slide content in a `figure` element.

Additionally, the Image component supports an `$html` property which can be used to add a `figcaption`, which is your HTML markup below.